src/core/net/dns_headers.hpp

```diff
@@ -341,6 +343,15 @@ class Header : public Clearable<Header>
      */
     void SetAdditionalRecordsCount(uint16_t aCount) { mArCount = HostSwap16(aCount); }
 
+#define GetZoneCount GetQuestionCount
+#define SetZoneCount SetQuestionCount
+#define GetPrerequisiteCount GetAnswerCount
+#define SetPrerequisiteCount SetAnswerCount
+#define GetUpdateCount GetAuthorityRecordsCount
+#define SetUpdateCount SetAuthorityRecordsCount
+#define GetAdditionalDataCount GetAdditionalRecordsCount
+#define SetAdditionalDataCount SetAdditionalRecordsCount
```

Comment:
Interesting defines 😃 

In client implementation I defined `UpdateHeader` as a sub-class of `Header` and defined these methods in it. 

I noticed that you included and squashed the DNS header related changes from the early branch https://github.com/abtink/openthread/commits/dns/new-rrs (the DNS header  RR changes) into the main server commit.

I included it in the client implementation as a seperate commit. There is also one for adding the log under SRP region.

I suggest rebasing the server code on top of those two commits, so we share the same common definitions and commit.  Thanks.

Reply:
@abtink  Thanks for the DNS records commits :+1: , will update the server to use the latest implementation.

> In client implementation I defined UpdateHeader as a sub-class of Header and defined these methods in it.

Will it increases code size? I was trying to avoid creating another class which has no custom logic.

Reply:
updated.

Reply:
I don't expect any code increase with `UpdateHeader`. It adds no member variables and all its methods are inlined.